File: spock/src/main/java/com/example/spock/member/MemberService.java
package com.example.spock.member;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
@RequiredArgsConstructor
public class MemberService {

  private final MemberRepository memberRepository;

  public Member createMember(final String name) {
    return memberRepository.save(new Member(name));
  }


}

File: spring-test/src/main/kotlin/com/spring/test/member/MemberService.kt
package com.spring.test.member

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class MemberService(
    private val memberRepository: MemberRepository
) {

    @Transactional
    fun create(name: String, email: String): Member {
        return memberRepository.save(Member(name, email))
    }
}